lib/Storage/MySqlTimeSeriesStore.php

```diff
@@ -0,0 +1,371 @@
+<?php
+/*
+ * Xibo - Digital Signage - http://www.xibo.org.uk
+ * Copyright (C) 2006-2015 Daniel Garner
+ *
+ * This file (PDOConnect.php) is part of Xibo.
+ *
+ * Xibo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Xibo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Xibo.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Xibo\Storage;
+
+use Xibo\Service\LogServiceInterface;
+
+/**
+ * Class MySqlTimeSeriesStore
+ * @package Xibo\Storage
+ */
+class MySqlTimeSeriesStore implements TimeSeriesStoreInterface
+{
+    /** @var StorageServiceInterface */
+    private $store;
+
+    /** @var LogServiceInterface */
+    private $log;
+
+    /**
+     * @inheritdoc
+     */
+    public function __construct($config = null)
+    {
+
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setDependencies($log)
+    {
+        $this->log = $log;
+        return $this;
+    }
+
+    /** @inheritdoc */
+    public function addMediaStat($statData)
+    {
+        $sql = 'INSERT INTO `stat` (`type`, statDate, start, `end`, scheduleID, displayID, layoutID, mediaID, Tag, `widgetId`) VALUES ';
+        $placeHolders = '(?, ?, ?, ?, ?, ?, ?, ?, ?, ?)';
+
+        $sql = $sql . implode(', ', array_fill(1, count($statData), $placeHolders));
+
+        // Flatten the array
+        $data = [];
+        foreach ($statData as $stat) {
+            foreach ($stat as $field) {
+                $data[] = $field;
+            }
+        }
+
+        $this->store->isolated($sql, $data);
+
+    }
+
+    /** @inheritdoc */
+    public function addLayoutStat($statData)
+    {
+        $sql = 'INSERT INTO `stat` (`type`, statDate, start, `end`, scheduleID, displayID, layoutID) VALUES ';
+        $placeHolders = '(?, ?, ?, ?, ?, ?, ?)';
+
+        $sql = $sql . implode(', ', array_fill(1, count($statData), $placeHolders));
+
+        // Flatten the array
+        $data = [];
+        foreach ($statData as $stat) {
+            foreach ($stat as $field) {
+                $data[] = $field;
+            }
+        }
+
+        $this->store->isolated($sql, $data);
+
+    }
+
+    /** @inheritdoc */
+    public function addTagStat($statData)
+    {
+
```

Comment:
Can we add a query for this too?

Reply:
We do not have tag stat records in MySQL. I need to pass tagStat array from doSubmitStats() of Soap.php. I can imagine the tagStat array as below: 

```
$tagStats[] = [
    'type' => $type,
    'statDate' => $now,
    'fromDt' => $fromdt,
    'toDt' => $todt,
    'scheduleId' => $scheduleId,
    'displayId' => $this->display->displayId,
    'tag' => $tag,
];
```

Reply:
Yes I think that is sensible.